### PR TITLE
[ANCHOR-505] Add fee details field to transaction object of SEP-6 and SEP-24

### PR DESCRIPTION
**Pull request by Ifropc**
## Background
Currently, both SEP-24 and SEP-6 offers an ability for anchor to charge fees. However, fees are only present to user as a single value (`transaction.fee`, without an ability to break it down. This proposal addresses this issue with ability to specify all charges being made (such as fees, taxes), on top of the existing summary value.  

## Proposed change
Deprecate `amount_fee` and `amount_fee_asset` in favor of a new field: `fee_details`. This field follows the same schema as fee details defined in SEP-38 standard.

### CODE REVIEW

File: ecosystem/sep-0024.md [hunk 1]
```diff
@@ -1158,6 +1159,22 @@ Each object in the `transactions` array should have the following fields:
...
 | `fee`     | string | The amount charged as a fee for processing the refund, in units of `amount_in_asset`.                                                                                                                                                                                |
 
+### Fee Details Object Schema
+
+| Name        | Type   | Description                                                                                                                                                           |
+| ----------- | ------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `total`     | string | The total amount of fee applied.                                                                                                                                      |
+| `asset`     | string | The asset in which the fee is applied, represented through the [Asset Identification Format](#asset-identification-format).                                           |
```
Comment (philipliu) [inline ecosystem/sep-0024.md:1167]: This should link to `sep-0038.md#asset-identification-format`. `#asset-identification-format` does not exist in this file. The same applies to SEP-6.

**Review by philipliu — APPROVED** [review]
LGTM, just a minor comment.

File: ecosystem/sep-0024.md [hunk 2]
```diff
@@ -1067,29 +1067,30 @@ Each object in the `transactions` array should have the following fields:
...
+| `amount_fee`              | string              | (**deprecated**) Amount of fee charged by anchor. This field is deprecated in favor of the `fee_details` field.                                                                                                                                                                                                                   |
+| `amount_fee_asset`        | string              | (**deprecated**, optional) The asset in which fees are calculated in. Must be present if the deposit/withdraw was made using non-equivalent assets. The value must be in [SEP-38 Asset Identification Format](sep-0038.md#asset-identification-format). See the [Asset Exchanges](#asset-exchanges) section for more information. |
+| `fee_details`             | object              | Description of fee charged by the anchor. The schema for this object is defined in the [Fee Details Object Schema](#fee-details-object-schema) section below.                                                                                                                                                                     |
```
Comment (JakeUrban) [inline ecosystem/sep-0024.md:1084]: Lets add a note here that `fee_details` should match the referenced quote's `fee` object.

**Review by JakeUrban — COMMENTED** [review]
I don't love that we're duplicating information on `GET /quote/:id` and `GET /transaction?id=`, is this avoidable? Am I correct in understanding that we're adding this because having fee info on the quote doesn't work when the quote is indicative or a quote is not used?

**Comment by Ifropc** [comment]
> Am I correct in understanding that we're adding this because having fee info on the quote doesn't work when the quote is indicative or a quote is not used?

Yes, you are right. I think duplicating fee object for when the quote is used is the better option

**Review by philipliu — DISMISSED** [review]
LGTM

File: ecosystem/sep-0006.md [hunk 3]
```diff
@@ -1462,6 +1463,22 @@ Each object in the `transactions` array should have the following fields:
...
 
+### Fee Details Object Schema
+
+| Name        | Type   | Description                                                                                                                                                           |
+| ----------- | ------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `total`     | string | The total amount of fee applied.                                                                                                                                      |
+| `asset`     | string | The asset in which the fee is applied, represented through the [Asset Identification Format](sep-0038.md#asset-identification-format).                                |
+| `breakdown` | array  | (optional) An array of objects detailing the fees that were used to calculate the conversion price. This can be used to datail the price components for the end-user. |
```
Comment (JakeUrban) [inline ecosystem/sep-0006.md:1472]: I know its redundant to have `fee_details.details` but I think its better than `breakdown`. Can we update the naming in this structure to match SEP-38's fee object as closely as possible?

**Review by JakeUrban — DISMISSED** [review]
Approved pending the comment, but I don't feel strongly about it.

File: ecosystem/sep-0006.md (hunk 3 above)
Comment (Ifropc) [reply ecosystem/sep-0006.md:1472]: That's a good point, I was going to reuse exact same object, but forgot to change the field name back

**Review by JakeUrban — APPROVED** [review]

**Pull request merged**